pkg/validator/server.go

```diff
@@ -98,27 +99,50 @@ func NewValidatingWebhook(webhookConfigName string, port int, certificater certi
 	return v, nil
 }
 
-// HandleValidation implements the HTTP API for the Validating Webhook.
-func (s *ValidatingWebhookServer) HandleValidation(w http.ResponseWriter, req *http.Request) {
-	defer dclose(req.Body)
-	adReq := new(admissionv1.AdmissionRequest)
-	if err := json.NewDecoder(req.Body).Decode(adReq); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		log.Error().Err(err).Msgf("Error reading admission request body; Responded to admission request with HTTP %v", http.StatusInternalServerError)
+func (s *ValidatingWebhookServer) doValidation(w http.ResponseWriter, req *http.Request) {
+	log.Trace().Msgf("Received validating webhook request: Method=%v, URL=%v", req.Method, req.URL)
+
+	if contentType := req.Header.Get(webhook.HTTPHeaderContentType); contentType != webhook.ContentTypeJSON {
+		err := errors.Errorf("Invalid content type %s; Expected %s", contentType, webhook.ContentTypeJSON)
+		http.Error(w, err.Error(), http.StatusUnsupportedMediaType)
+		log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusUnsupportedMediaType)
 		return
 	}
 
-	data, err := json.Marshal(s.handleValidation(adReq))
+	admissionRequestBody, err := webhook.GetAdmissionRequestBody(w, req)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		log.Error().Err(err).Msgf("Error marshaling admission response body; Responded to admission request with HTTP %v", http.StatusInternalServerError)
+		// Error was already logged and written to the ResponseWriter
```

Comment:
Is there any particular reason `GetAdmissionRequest()` needs to log the error and also return it? Not that there's anything wrong with this, just sticks out as a bit odd. I'd expect the error to be logged here vs. inside the function.

Reply:
`GetAdmissionRequestBody()` logs an error and writes an error response on the http server. It seems easier to write the HTTP response error and log it in the same place, for context.